src/Messages/TSMessagesManager.m

```diff
@@ -582,7 +582,8 @@ - (void)handleEndSessionMessageWithEnvelope:(OWSSignalServiceProtosEnvelope *)en
     }];
 
     dispatch_async([OWSDispatch sessionStoreQueue], ^{
-        [[TSStorageManager sharedManager] deleteAllSessionsForContact:endSessionEnvelope.source];
+        [[TSStorageManager sharedManager] deleteSessionForContact:endSessionEnvelope.source
+                                                         deviceId:endSessionEnvelope.sourceDevice];
```

Comment:
Right, fixes the scenario wherein:

* Alice.1 has a session with Bob.1 and Bob.2
* Bob.1 sends an EndSession to Alice
* Thus Alice.1 deletes her sessions with Bob.1 and Bob.2
* Bob.2 sends Alice1 "hi" (with the existing session - as a regular WM, not a PKWM)
* Alice has no session with Bob.2 so cannot decrypt the message.